### PR TITLE
Add Background Color option to Media and Text widget

### PR DESCRIPTION
**Pull request by tqureshi-uog**
# Summary of changes
Add Background Color option to Media and Text widget so they can be used instead of custom-built YAML components

## Frontend
- add relevant background color fields to `gatsby-node.js` schema
- add background and text color logic to `mediaText.js`
- add `data-title` fields to `widgets.js` and `sectionWidgets.js` for reference purposes (unlike the html `title` attribute, `data-title` does not result in a tooltip on hover and is invisible to end users unless they view source code)
- remove obsolete media links field
- upgrade packages

## Backend
- add Background Color taxonomy with choices based on Bootstrap 5 classes
- add `Background` field to Media and Text paragraph 
- remove obsolete links field from Media and Text paragraph

# Test Plan

- Go to https://tqtest.gatsbyjs.io/media-text-testing and ensure it's consistent with the original version on https://preview-ugconthub.gtsb.io/media-text-testing
- Go to https://tqtest.gatsbyjs.io/media-text-test2 and ensure the colors display correctly (headings and text should be black for uog-blue-muted and light-gray backgrounds, white for dark-gray backgrounds, and default otherwise, i.e. dark text, red heading)
- Do the same for https://tqtest.gatsbyjs.io/media-text-video
- Review https://tqtest.gatsbyjs.io/bcomm/become-a-global-leader and ensure the two `Future You` media/text widgets have the uog-blue-muted background color
- Review https://tqtest.gatsbyjs.io/study-in-canada and verify the `Things You Should Know About the City of Guelph` media and text widgets look and behave like the custom YAML widgets on https://www.uoguelph.ca/study-in-canada
- Drupal multidev can be reviewed at https://medtxtbg-chug.pantheonsite.io/

### CODE REVIEW

**Comment by tqureshi-uog** [comment]
> The page https://tqtest.gatsbyjs.io/media-text2 doesn't seem to exist.

Sorry, typo - the link is https://tqtest.gatsbyjs.io/media-text-test2

**Comment by mmafe** [comment]
I find the bg-light option so light that I can barely tell the difference between that and the white background. I think we should just limit the options to the light-blue (which will be lighter than it currently is for accessibility) and the dark grey.
image


Although I worry that we're adding a can of worms by even adding the dark grey option. I like the way its looks, however there may be additional accessibility issues with it. The red headings for example fail colour contrast.
image

**Comment by abanuog** [comment]
Agree with Miranda that the bg-light option could be removed.

File: src/components/shared/mediaText.js [hunk 1]
```diff
@@ -30,9 +30,36 @@ function MediaText (props) {
...
+    let textColBg;
+    let textColPadding;
     let wrapperCol;
     let headingClass;
+    let headingColor;
     let textOrButtons = mediaDescription || mediaButtons ? true : false;
+    
+    switch(mediaBgColor) {
+        case "Blue Muted":
+            textColBg = " uog-blue-muted";
```
Comment (mmafe) [inline src/components/shared/mediaText.js:42]: I feel like someday a person could easily forget to include that first space. Could we use the [classnames](https://www.npmjs.com/package/classnames) package in the rendering of the classnames so we don't have to worry about spaces?

**Comment by abanuog** [comment]
I think you have to use single quotes in the classNames utility

**Comment by tqureshi-uog** [comment]
> I think you have to use single quotes in the classNames utility

Changed the syntax but it didn't make a difference - the problem was due to another issue which should now be resolved.

File: src/components/shared/mediaText.js (hunk 1 above)
Comment (tqureshi-uog) [reply src/components/shared/mediaText.js]: Done

**Review by mmafe — APPROVED** [review]
Admittedly, I'm a little nervous that the bg-dark option is going to have inaccessible use cases that we haven't thought about. (And you're right that there are a ton of !importants in the code.)

That being said, this seems like one of those moments where the styling is a temporary situation until a new stylesheet can handle these kinds of colour contrast fixes from the ground up rather than through overrides.

If we can fix the primary-outline and info-outline colour contrast, that would be great. Other than that, not seeing roadblocks.

**Comment by tqureshi-uog** [comment]
> If we can fix the primary-outline and info-outline colour contrast, that would be great. Other than that, not seeing roadblocks.

Done - see latest changes on https://tqtest.gatsbyjs.io/media-text-test2

**Pull request merged**